sycl/doc/extensions/proposed/sycl_ext_intel_user_defined_reductions.asciidoc

```diff
@@ -0,0 +1,138 @@
+= sycl_ext_intel_user_defined_reductions
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Notice
+
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc. OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+NOTE: This document is better viewed when rendered as html with asciidoctor.
+GitHub does not render image icons.
+
+This extension is written against the SYCL 2020 revision 5 specification. All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+**TODO**: dependency on `group_with_scrathpad` from `sycl_ext_oneapi_group_sort`
+
+NOTE: THis extension is experimental: interfaces are subject to change later.
+
+== Introduction
+
+The purpose of this extension is to expand functionality of `reduce_over_group`
+free function defined in section 4.17.4.5. `reduce` of the core SYCL
+specification by allowing to pass user-defined binary operators in there as well
+as perform reductions on a non-fundamental types.
+
+== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 Feature test macros. Therefore, an implementation
+supporting this extension must predefine the macro
+`SYCL_EXT_INTEL_USER_DEFINED_REDUCTIONS` to one of the values defined in the
```

Comment:
```suggestion
`SYCL_EXT_ONEAPI_USER_DEFINED_REDUCTIONS` to one of the values defined in the
```

Reply:
Applied in 26b93170f6afa6dca47f62bf487b65e5854e0c54